File: 0ld/hackerrank/jumping_clouds.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    int n(0), i(0);
    cin >> n;
    int clouds[n];
    for(i = 0; i < n; i++){
        cin >> clouds[i];
    }
    int jumps(0);
    for(i = 0; i < n-2;){
        if(clouds[i+2] == 0){
            jumps++;
            i+=2;
            cout << 'i' << i << endl;
            continue;
        } else if(clouds[i+1] == 0){
            jumps++;
            i++;
            cout << 'i' << i << endl;
        }
    }
    cout << jumps << endl;
    return 0;
}

File: 0ld/hackerrank/jumping_clouds.py
def get_solution(pos, clouds, jumps):
    if(pos == len(clouds) - 3):
        if(clouds[pos + 2] == 0):
            return jumps+1
    elif(pos == len(clouds) - 2):
        if(clouds[pos+1] == 0):
            return jumps+1
    else:
        if(clouds[pos+2] == 0):
            return get_solution(pos+2, clouds, jumps+1)
        elif(clouds[pos+1] == 0):
            return get_solution(pos+1, clouds, jumps+1)


n = int(input())
clouds = list(map(int, input().split()))
jumps = 0
print(get_solution(0, clouds, 0))
